packages/contracts/contracts/minter-suite/Minters/MinterSetPriceV5Holder.sol

```diff
@@ -0,0 +1,795 @@
+     * projects whose holders are allowed to purchase a token on `projectId`
+     */
+    mapping(address => mapping(uint256 => mapping(address => mapping(uint256 => bool))))
+        public allowedProjectHolders;
+
+    // function to restrict access to only AdminACL allowed calls
+    // @dev defers which ACL contract is used to the core contract
+    function _onlyCoreAdminACL(
+        address _coreContract,
+        bytes4 _selector
+    ) internal {
+        require(
+            IGenArt721CoreContractV3_Base(_coreContract).adminACLAllowed(
+                msg.sender,
+                address(this),
+                _selector
+            ),
+            "Only Core AdminACL allowed"
+        );
+    }
+
+    function _onlyArtist(
+        uint256 _projectId,
+        address _coreContract
+    ) internal view {
+        require(
+            msg.sender ==
+                IGenArt721CoreContractV3_Base(_coreContract)
+                    .projectIdToArtistAddress(_projectId),
+            "Only Artist"
+        );
+    }
+
+    /**
+     * @notice Initializes contract to be a Filtered Minter for
+     * `_minterFilter`, integrated with Art Blocks core contract
+     * at address `_genArt721Address`.
+     * @param _minterFilter Minter filter for which
+     * this will a filtered minter.
+     * @param _delegationRegistryAddress Delegation registry contract address.
+     */
+    constructor(
+        address _minterFilter,
+        address _delegationRegistryAddress
+    ) ReentrancyGuard() {
+        delegationRegistryAddress = _delegationRegistryAddress;
+        emit DelegationRegistryUpdated(_delegationRegistryAddress);
+        delegationRegistryContract = IDelegationRegistry(
+            _delegationRegistryAddress
+        );
+        minterFilterAddress = _minterFilter;
+        minterFilter = IMinterFilterV1(_minterFilter);
+    }
+
+    /**
+     * @notice Returns whether or not the provided address `_coreContract`
+     * is an Art Blocks Engine core contract. Caches the result for future access.
+     * @param _coreContract Address of the core contract to check.
+     */
+    function _isEngine(address _coreContract) internal returns (bool) {
+        SplitFundsLib.IsEngineCache storage isEngineCache = _isEngineCaches[
+            _coreContract
+        ];
+        if (isEngineCache.isCached) {
+            return isEngineCache.isEngine;
+        } else {
+            bool isEngine = SplitFundsLib.getV3CoreIsEngine(
+                _coreContract,
+                isEngineCache
+            );
+            return isEngine;
+        }
+    }
+
+    /**
+     *
+     * @notice Registers holders of NFTs at address `_NFTAddress` to be
+     * considered for minting. New core address is assumed to follow syntax of:
+     * `projectId = tokenId / 1_000_000`
+     * @param _NFTAddress NFT core address to be registered.
+     */
+    function registerNFTAddress(
```

Comment:
Since this is a contract-level minter setting, we don't have a way to currently index this. I think we had discussed just removing this constraint altogether, and settled on removing this guardrail and relying on the artist to only allowlist appropriate projects. Does that sound good to you? (same goes for `unregisterNFTAddress`)